### PR TITLE
Feat: Add endpoint to move to ready

### PR DESCRIPTION
**Pull request by biruwon**
Will move deploy to ready state when in validation

### CODE REVIEW

File: swagger.yml [hunk 1]
```diff
@@ -1076,6 +1076,22 @@ paths:
...
+    post:
+      operationId: readyDeploy
+      tags: [deploy]
+      parameters:
+        - name: deploy_id
+          type: string
+          in: path
+          required: true
+      responses:
+        '201':
```
Comment (biruwon) [inline swagger.yml:1089]: Not sure why on [the BB side](https://github.com/netlify/bitballoon/pull/16569/files#diff-c6ed00a289567dfc399f1032e78413680734d0cb65230b37df98d72e615f4663R368) we are saying a 200 for this and the rest of operations that update the deploy state

File: swagger.yml [hunk 2]
```diff
@@ -1076,6 +1076,22 @@ paths:
             $ref: '#/definitions/deploy'
         default:
           $ref: '#/responses/error'
+  /deploys/{deploy_id}/ready:
```
Comment (4xposed) [inline swagger.yml:1079]: I'm not sure if it has already been discussed but to me this does not seem very RESTful.

What would you think to use the `update` endpoint `PUT /deploys/{deploy_id}` with a payload of `{state: "READY"}`

we could defer by the payload what to do, and not only seems a bit more RESTful it's a bit more representative of the change we want to do with the resource (Deploy)

I wonder what the rest of the team would think of this as well.

cc. @JGAntunes @kitop @julienp

Comment (kitop) [reply swagger.yml:1079]: Moving this convo to https://github.com/netlify/bitballoon/pull/16569/files#r1315114299

File: swagger.yml (hunk 1 above)
Comment (kitop) [reply swagger.yml:1089]: I don't quite follow. 200 seems the correct response in most cases. Could be a copy-pasta error in others? Seems like we should respect 200 here, right?

Comment (biruwon) [reply swagger.yml]: It seems to be the case, updated ready, cancel and restore endpoint to match what we really return, as 201 should be when a resource it's created.

**Comment by kitop** [comment]
Closing this based on the chats we've been having.

**Pull request closed without merging**